packages/web-api-scan-runner/src/runner/runner.ts

```diff
@@ -0,0 +1,138 @@
+import { ScannerTask } from '../tasks/scanner-task';
+import { WebDriverTask } from '../tasks/web-driver-task';
+
+@injectable()
+export class Runner {
+    constructor(
+        @inject(GuidGenerator) private readonly guidGenerator: GuidGenerator,
+        @inject(ScanMetadataConfig) private readonly scanMetadataConfig: ScanMetadataConfig,
+        @inject(ScannerTask) private readonly scannerTask: ScannerTask,
+        @inject(OnDemandPageScanRunResultProvider) private readonly onDemandPageScanRunResultProvider: OnDemandPageScanRunResultProvider,
+        @inject(WebDriverTask) private readonly webDriverTask: WebDriverTask,
+        @inject(Logger) private readonly logger: Logger,
+        @inject(PageScanRunReportService) private readonly pageScanRunReportService: PageScanRunReportService,
+        private readonly convertAxeToSarifFunc = convertAxeToSarif,
+    ) {}
+
+    public async run(): Promise<void> {
+        let browser: Browser;
+        const scanMetadata = this.scanMetadataConfig.getConfig();
+        let pageScanResult: OnDemandPageScanResult;
+
+        this.logger.logInfo(`Reading Page Scan Run id= ${scanMetadata.id}`);
+        pageScanResult = (await this.onDemandPageScanRunResultProvider.readScanRuns([scanMetadata.id]))[0];
+
+        // set scanned page run state to running
+        pageScanResult.run = this.getRunResult('running');
+
+        this.logger.logInfo(`Updating page scan to running`);
+        await this.onDemandPageScanRunResultProvider.updateScanRun(pageScanResult);
+        pageScanResult = (await this.onDemandPageScanRunResultProvider.readScanRuns([scanMetadata.id]))[0];
+
+        try {
+            // start new web driver process
+            browser = await this.webDriverTask.launch();
+
+            await this.scan(pageScanResult);
+        } catch (error) {
+            this.logger.logInfo(`Scan failed ${error}`);
+            pageScanResult.run = this.getRunResult('failed', error instanceof Error ? error.message : `${error}`);
+            pageScanResult.scanResult = { state: 'unknown', issueCount: 0 };
+            pageScanResult.reports = [];
+        } finally {
+            try {
+                await this.webDriverTask.close();
+            } catch (error) {
+                this.logger.logError(`unable to close web driver ${error}`);
+            }
+        }
+
+        this.logger.logInfo(`Updating page scan in database`);
+        await this.onDemandPageScanRunResultProvider.updateScanRun(pageScanResult);
+    }
+
+    private async scan(pageScanResult: OnDemandPageScanResult): Promise<void> {
+        this.logger.logInfo(`Running Scan`);
+
+        const axeScanResults = await this.scannerTask.scan(pageScanResult.url);
+
+        if (!isNil(axeScanResults.error)) {
+            this.logger.logInfo(`Changing page status to failed`);
+            pageScanResult.run = this.getRunResult('failed', axeScanResults.error);
+            pageScanResult.scanResult = { state: 'unknown', issueCount: 0 };
```

Comment:
similar to above no need to set scanResult and reports on failure

Reply:
similar reason